src/core/load-fragment.c

```diff
@@ -3539,6 +3540,80 @@ int config_parse_restrict_namespaces(
 }
 #endif
 
+int config_parse_restrict_filesystems(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        ExecContext *c = data;
+        bool invert = false;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                /* Empty assignment resets the list */
+                c->restrict_filesystems = set_free(c->restrict_filesystems);
+                c->restrict_filesystems_allow_list = false;
+                return 0;
+        }
+
+        if (rvalue[0] == '~') {
+                invert = true;
+                rvalue++;
+        }
+
+        if (!c->restrict_filesystems) {
+                c->restrict_filesystems = set_new(NULL);
```

Comment:
so you are usual "trivial_hash_ops" here, i.e. the set object is keyed by hashed pointer values, not by the strings they point to. This is weird, what's the point? You want to pass `string_hash_ops` or so here.

But actually, don't allocate this here at all. The nice thing about set_put_strdup() is that it allocates the set if not allocated yet, i.e. lazily, and implicitly uses the string hash ops for that with destructors. so you can shorten this quite a bit